tests/D2L.CodeStyle.TestAnalyzers.Tests/TestCaseData/TestCaseDataAnalyzerTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using D2L.CodeStyle.TestAnalyzers.Test.Verifiers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace D2L.CodeStyle.TestAnalyzers.TestCaseData {
+
+	internal sealed class TestCaseDataAnalyzerTests : DiagnosticVerifier {
+		protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer() {
+			return new TestCaseDataAnalyzer();
+		}
+
+		[Test]
+		public void EmptyDocument_NoDiag() {
+			const string test = @"";
+
+			VerifyCSharpDiagnostic( test );
+		}
+
+		[Test]
+		public void DocumentWithoutThrows_NoDiag() {
+			const string test = @"
+	using System;
+
+	namespace test {
+		class Test {
+
+			[Test]
+			public void TestWithoutThrows() {
+			}
+
+		}
+	}";
+			AssertNoDiagnostic( test );
+		}
+
+		[Test]
+		public void DocumentWithThrowsCase1_Diag() {
+			const string test = @"
+	using System;
+
+	namespace test {
+		class Test {
+			private TestCaseData[] UserDetailsTestSource = {
+				new TestCaseData(0,0 ).SetCategory('').SetName( 'NoExistingResult_500Ids_Success' ).Throws( typeof( Exception ) ),

```

Comment:
The `'` is not valid here; those are characters, not strings.
- `SetCategory('')` no such method
- `SetName( 'NoExistingResult_500Ids_Success' )` not valid C# code (multicharacter character)
- `Throws('')` no such method